src/cloudimagedirectory/transform/transform.py

```diff
@@ -290,4 +290,22 @@ def run(self, data):
 
         results.sort()
 
-        return [connection.DataEntry("idx/list/image-names", results)]
+        return [connection.DataEntry("v1/idx/list/image-names", results)]
+
+
+class TransformerV2All(Transformer):
+    """Genearate list of all image details."""
+
+    def run(self, data):
+        """Sort the raw data."""
+        # NOTE: Verify that the data is not raw.
+        entries = [x for x in data if not x.is_raw() and not x.is_provided_by("idx")]
+
+        results = []
+
+        for entry in entries:
+            results.append(entry.content)
+
+        results.sort(key=lambda x: x["name"], reverse=False)
+
+        return [connection.DataEntry("v2/all", results)]
```

Comment:
Oh, so it is for all images from all cloud providers. Won't it be confusing? These data structures will always look a little different (or at least contain different types of values). It also means that the image data have to be extended, because (from what I remember) they don't contain the cloud provider, the region and sometimes the architecture.

Reply:
> It also means that the image data have to be extended, because (from what I remember) they don't contain the cloud provider, the region and sometimes the architecture.

That is a valid point. I started with a first PR of two, which divides this into two parts. The first will remove the version flag and add a version prefix to the reference path. The 2nd one will then extend our schema by region, cloud provider and whatever else is missing. I will double check :smiley: 
1. https://github.com/redhatcloudx/transformer/pull/585